pandas/util/decorators.py

```diff
@@ -233,3 +233,24 @@ def make_signature(func):
     if spec.keywords:
         args.append('**' + spec.keywords)
     return args, spec.args
+
+
+def generate_dynamic_docstring(creator):
+    """
+    this function returns a dynamically generated doc-string
+    that will call the creator callable to actually
+    create the docstring.
```

Comment:
@jorisvandenbossche this will dynamically create the docstrings

though I did this by experimentation (e.g. its code in ``inspect.py`` which actually gets the docs)